colearn_keras/keras_learner.py

```diff
@@ -76,13 +76,46 @@ def __init__(self, model: keras.Model,
 
         self.vote_score: float = self.test(self.train_loader)
 
+    def recompile_model(self):
+        """
+        Recompiles the Keras model. This way the optimizer history get erased,
+        which is needed before a new training round, otherwise the outdated history is used.
+        """
+        compile_args = self.model._get_compile_args()  # pylint: disable=protected-access
```

Comment:
I did not find any better way of resetting the optimizer, than re-compiling the mode, which needs calling [this protected function](https://github.com/tensorflow/tensorflow/blob/ec00b562a4e64864cbb4311b3d9ffd97ea4bd22e/tensorflow/python/keras/engine/training.py#L2768).

When I simply replace the optimizer
```python
new_opt = keras.optimizers.SGD(lr=0.1)
model.optimizer = new_opt
```
then I cannot change batch size anymore, and also the optimizer seems not to be updated, the `model.optimizer.get_weights()` should store the number of steps and the momentums, but it remains an empty list this way.

Reply:
maybe name the function `reset_optimizer` to better reflect why we have this and when we want to use it